gen/flights/flightplan.py

```diff
@@ -1182,14 +1182,27 @@ def racetrack_for_objective(
         start = end.point_from_heading(heading - 180, diameter)
         return start, end
 
-    @staticmethod
-    def aewc_orbit(location: MissionTarget) -> Point:
-        closest_airfield = location
-        # TODO: This is a heading to itself.
-        # Place this either over the target or as close as possible outside the
-        # threat zone: https://github.com/Khopa/dcs_liberation/issues/842.
-        heading = location.position.heading_between_point(closest_airfield.position)
-        return location.position.point_from_heading(heading, 5000)
+    def aewc_orbit(self, location: MissionTarget) -> Point:
+        heading = location.position.heading_between_point(location.position)
+
+        position = ShapelyPoint(
+            self.package.target.position.x, self.package.target.position.y
+        )
+
+        # in threat zone
+        if meters(position.distance(self.threat_zones.all)) == meters(0):
```

Comment:
`if self.threat_zones.threatened(position)`

If you do that you also don't need to explicitly make the `ShapelyPoint`, it'll accept `location.position` so you can delete L1188-L1190.